src/unix/fs.c

```diff
@@ -1310,6 +1311,29 @@ static ssize_t uv__fs_copyfile(uv_fs_t* req) {
     }
   }
 
+  /**
+   * Change the timestamps of the destination file to match the source file.
+   */
+#if defined(__APPLE__)
+  times[0] = src_statsbuf.st_atimespec;
+  times[1] = src_statsbuf.st_mtimespec;
+#else
+  times[0] = src_statsbuf.st_atim;
+  times[1] = src_statsbuf.st_mtim;
+#endif
+
+  if (utimensat(AT_FDCWD, req->new_path, times, 0) == -1) {
+    err = UV__ERR(errno);
+    goto out;
+  }
+
+  /*
+   * Change the ownership and permissions of the destination file to match the
+   * source file.
+   * `cp -p` does not care about errors here, so we don't either.
+   */
+  if (chown(req->new_path, src_statsbuf.st_uid, src_statsbuf.st_gid) == -1) {}
```

Comment:
It looks like the surrounding code also prefers operations on the file descriptor rather than the path.

Reply:
True that, fchown is probably a better choice.

Reply:
Will do.

Reply:
Fixed in https://github.com/libuv/libuv/commit/a5c5097d15ea8c8840656ae677b752e4fcb2c28b